### PR TITLE
Add vmap support for rng_bit_generator.

### PR DESCRIPTION
**Pull request by LenaMartens**
Adds a naive batching rule which loops over the primitive, then stacks the results. This has the advantage that the vmapped result will be the same as calling the primitive n times, but is not very performant.

### CODE REVIEW

**Review by froystig — APPROVED** [review]
This is an "inline" loop that will emit N individual primitive calls. Is there a batch dimension length at which `jax.lax.map` would make sense instead? We can follow up after this PR if there's an answer to that question.

**Comment by froystig** [comment]
Should we give some kind of warning in the meantime, especially if the batch dimension is large?

**Comment by froystig** [comment]
I'm not sure what's "too large" but it might depend on the backend/device. Conceptually we want a `map` and that alone might be reason enough to switch. One thing we could do is always use an explicit `scan` for the map and then tune its `unroll` parameter for performance. This has the additional advantage of not increasing the jaxpr size, only the HLO.
